src/Uri.php

```diff
@@ -585,11 +585,14 @@ private function filterUserInfoComponent($component): string
             throw new \InvalidArgumentException('User info must be a string');
         }
 
-        return preg_replace_callback(
+        /** @var string $thing */
```

Comment:
No need to do an assignment here. Just add `/** @var string */` above the return statement.